### ISSUE TITLE
Icons not populating, name changes to 'fa-undefined' or 'NaN'

### ISSUE THREAD

**Issue by sclanc** [travelist/angular2-fontawesome#13]
angular2-fontawesome works, but Icons are being populated in the DOM as 0x0 pixel fa elements with a name property of either  'fa-undefined' or 'NaN'.  I have tried using both directives and components with no success. I have also tried to duplicate every example in the documentation, and still can't get it to work.

boilerplate: angular-cli

system-config

```
const map: any = {
   "materialize": "vendor/materialize-css",
   "angular2-materialize": "vendor/angular2-materialize",
   "jquery": "vendor/jquery",
   "angular2-fontawesome": "vendor/angular2-fontawesome",
};

/** User packages configuration. */
const packages: any = {
  'materialize': {
    "format": "global",
    "main": "dist/js/materialize",
    "defaultExtension": "js"
  },
  'angular2-materialize': {
    "main": "dist/index",
    "defaultExtension": "js"
  },
   'angular2-fontawesome': {
     "defaultExtension": 'js' }
}; 
```

app.component.ts 

```
import { Component } from '@angular/core';
import {MaterializeDirective} from "angular2-materialize";
import { ROUTER_DIRECTIVES } from '@angular/router';
import { FaComponent } from 'angular2-fontawesome/components';
import { FaDirective } from 'angular2-fontawesome/directives';

@Component({
  moduleId: module.id,
  selector: 'app-root',
  templateUrl: 'app.component.html',
  styleUrls: ['styles/app.component.css'],
  directives: [MaterializeDirective, ROUTER_DIRECTIVES, FaComponent, FaDirective]
})
export class AppComponent {}
```

package.json (not complete)

```
 "dependencies": {
    "@angular/common": "2.0.0-rc.4",
    "@angular/compiler": "2.0.0-rc.4",
    "@angular/core": "2.0.0-rc.4",
    "@angular/forms": "0.2.0",
    "@angular/http": "2.0.0-rc.4",
    "@angular/platform-browser": "2.0.0-rc.4",
    "@angular/platform-browser-dynamic": "2.0.0-rc.4",
    "@angular/router": "3.0.0-beta.2",
    "angular2-fontawesome": "^0.5.1",
    "angular2-materialize": "^3.0.3",
```

angular-cli-build.js 

```
var Angular2App = require('angular-cli/lib/broccoli/angular2-app');

module.exports = function(defaults) {
  return new Angular2App(defaults, {
    vendorNpmFiles: [
      'systemjs/dist/system-polyfills.js',
      'systemjs/dist/system.src.js',
      'zone.js/dist/**/*.+(js|js.map)',
      'es6-shim/es6-shim.js',
      'reflect-metadata/**/*.+(ts|js|js.map)',
      'rxjs/**/*.+(js|js.map)',
      '@angular/**/*.+(js|js.map)',
      'jquery/dist/*',
      'angular2-materialize/dist/*',
      'materialize-css/dist/**/*',
      'angular2-fontawesome/**/*'
    ]
  });
};
```

Thanks a bunch.

**Comment by travelist** [comment]
@sclanc Thank you for opening the issue:)

Would you try to write like this in your `app.component.html`

```
<fa [name]="rocket" [border]=true></fa>
```

Or please try this

```
<fa [name]="'rocket'" [border]=true></fa>
```

fa-`undefined` might come from the `[name]` attribute.

**Comment by sclanc** [comment]
Alright, I gave that I try, here's what I've got: 

app.component.html 

```
     <footer class="page-footer indigo darken-1">
      <div class="container">
        <div class="row">
          <div class="center">
            <fa [name]="rocket" [border]=true></fa>
            <fa [name]="'rocket'" [border]=true></fa>
          </div>
        </div>
      </div>
      <div class="footer-copyright">
        <div class="container">
        Built with Angular2 JS
        </div>
      </div>
    </footer>
```

Here is what the HTML is looking like after it is rendered: 

```
<fa _ngcontent-lew-1="" ng-reflect-border="true"><i ng-reflect-raw-class="fa,fa-undefined,fa-border" class="fa fa-undefined fa-border"></i></fa>
<fa _ngcontent-lew-1="" ng-reflect-name="rocket" ng-reflect-border="true"><i ng-reflect-raw-class="fa,fa-rocket,fa-border" class="fa fa-rocket fa-border"></i></fa>
```

As you can see, the second instance of the element is not undefined, however it is still not appearing on the web page, when inspecting the icon it says " 1.fa.fa-rocket.fa-border | 0x0 

Also, thanks for your quick response.

**Comment by travelist** [comment]
@sclanc then, `font-awesome` itself might not be included.

Would you check the `angular-cli-build.js` file having `font-awesome/css/*.*` and  `font-awesome/fonts/*.*` in `vendorNpmFiles` ?

I guess this might solve this issue

``` javascript
module.exports = function(defaults) {
  return new Angular2App(defaults, {
    vendorNpmFiles: [
      ...
      // Add following lines (1/2) actually not sure whether all the lines are needed or not.
      'angular2-fontawesome/*.+(js|js.map)',
      'angular2-fontawesome/**/*.+(js|js.map)',
      'angular2-fontawesome/**/**/*.+(js|js.map)',

      // You need to add following lines as well (2/2)
      'font-awesome/css/*.*',
      'font-awesome/fonts/*.*'
    ]
```

**Comment by sclanc** [comment]
Here is what my angular-cli-build.js looks like now: 

```
// Angular-CLI build configuration
// This file lists all the node_modules files that will be used in a build
// Also see https://github.com/angular/angular-cli/wiki/3rd-party-libs

/* global require, module */

var Angular2App = require('angular-cli/lib/broccoli/angular2-app');

module.exports = function(defaults) {
  return new Angular2App(defaults, {
    vendorNpmFiles: [
      'systemjs/dist/system-polyfills.js',
      'systemjs/dist/system.src.js',
      'zone.js/dist/**/*.+(js|js.map)',
      'es6-shim/es6-shim.js',
      'reflect-metadata/**/*.+(ts|js|js.map)',
      'rxjs/**/*.+(js|js.map)',
      '@angular/**/*.+(js|js.map)',
      'jquery/dist/*',
      'angular2-materialize/dist/*',
      'materialize-css/dist/**/*',
      'angular2-fontawesome/*.+(js|js.map)',
      'angular2-fontawesome/**/*.+(js|js.map)',
      'angular2-fontawesome/**/**/*.+(js|js.map)',
      'font-awesome/css/*.*',
      'font-awesome/fonts/*.*'
    ]
  });
};
```

The same behavior persists.

**Comment by travelist** [comment]
@sclanc Maybe restart the process..? And in `app.component.ts`, 

Following line might be needed

```
styleUrls: ['vendor/font-awesome/css/font-awesome.css']
```

in our case,

Change 

```
styleUrls: ['styles/app.component.css']
```

to

```
styleUrls: ['styles/app.component.css', 'vendor/font-awesome/css/font-awesome.css']
```

**Comment by sclanc** [comment]
Sorry, I should have included that I did restart the process, I restarted it one more time just for good measure, I also cleared the cache.

**Comment by travelist** [comment]
the problem might be `font-awesome` icons (not `angular2-fontawesome`) is not included to our app.

Please check following line can render the fa icon

```
<i class="fa fa-rocket"></i>
```

if not render the icon,
1. ensure `fant-awesome` in `vendor` directory
2. check `styleUrls` have `font-awesome`

```
styleUrls: ['styles/app.component.css', 'vendor/font-awesome/css/font-awesome.css']
```

**Comment by sclanc** [comment]
Alright, I this is what my app.components.ts looks like now: 

```
import { Component } from '@angular/core';
import {MaterializeDirective} from "angular2-materialize";
import { ROUTER_DIRECTIVES } from '@angular/router';
import { FaComponent } from 'angular2-fontawesome/components';
import { FaDirective } from 'angular2-fontawesome/directives';

@Component({
  moduleId: module.id,
  selector: 'app-root',
  templateUrl: 'app.component.html',
  styleUrls: ['styles/app.component.css', '../vendor/font-awesome/css/font-awesome.css'],
  directives: [MaterializeDirective, ROUTER_DIRECTIVES, FaComponent, FaDirective]
})
export class AppComponent {}
```

I had to modify the path slightly because of the moduleID, but I am still seeing the same thing. I also tried changing the size of the icons (just to see if maybe they were there?)  with no success.  I will add that trying to add the regular icon without the angular2 font awesome directive made a black rectangle appear  , so I am guessing that the issue is that the app can't access the icons, font-awesome is in my node_modules folder.

**Comment by sclanc** [comment]
UPDATE: I think I know what the problem is!  Angular2-fontawesome is trying to get font-awesome from 

`http://localhost:4200/fonts/fontawesome-webfont.woff2?v=4.6.3`

but it should be looking in 

`http://localhost:4200/vendor/font-awesome/fonts/fontawesome-webfont.woff2?v=4.6.3`

 so the css isn't getting accessed because of bad calls. I tried updating the system.config to fix the issue but it didn't work, I'm not sure where to go from here.

**Comment by sclanc** [comment]
It's a little hacky, but I got the icons to appear by copying the fonts folder from node_modules/font-awesome to my src folder. However, the directive and components still don't work.

**Comment by travelist** [comment]
@sclanc Sorry I could not check github yesterday.. and really thank you for these comments. Appreciate!

I tried to reproduce the error and I've hit the same error with `FaComponent` (somehow `FaDirective` works correctly..) I made `demo` branch and uploaded the code there.

<img width="195" alt="screen shot 2016-07-30 at 8 07 30 pm" src="https://cloud.githubusercontent.com/assets/3145993/17272609/d5444dfe-5692-11e6-8bb0-9a2666e9611b.png">

For now I might not possibly make spare time, but will fix it at some point of the time (may be when the angular2 version is updated..) 

Anyway, Thanks you very much for your help!!:)
